proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1645 @@
+
+```scss
+$pink: oklch(64% 0.196 353); // hsl(329.8 70.29% 58.75%)
+$blue: oklch(64% 0.196 253); // hsl(207.4 99.22% 50.69%)
+```
+
+The new `color()` function provides access to a number of specialty spaces.
+Most notably, `display-p3` is a common space for wide-gamut monitors, making
+it likely one of the more popular options for authors who simply want access to
+a wider range of colors. For example, P3 greens are significantly 'brighter'
+and more saturated than the greens available in sRGB:
+
+```scss
+$fallback-green: rgb(0% 100% 0%);
+$brighter-green: color(display-p3 0 1 0);
+```
+
+<!-- flesh this out more -->
+
+### Design Decisions
+
+Most of the design decisions involved in the proposal are based on the
+[CSS Color Level 4][color-4] specification, which we have tried to emulate as
+closely as possible, while maintaining support for legacy projects. In some
+cases, that required major changes to the way Sass handles colors:
+
+1. Channel values are no longer clamped to the gamut of a given color space.
+   Authors can use the provided `to-gamut()` function to map colors into gamut,
+   or allow out-of-gamut colors in output. Browsers will also provide improved
+   gamut mapping, based on the user device capabilities.
+2. Channel values are no longer rounded to the nearest integer, since the spec
+   now requires maintaining precision wherever possible. This is especially
+   important in RGB spaces, where color distribution is inconsistent.
+
+While we are not attempting to support all of [CSS Color Level 5][color-5] at
+this point, we have used it as a reference while updating color manipulation
+functions such as `color.mix()`.
+
+[color-5]: https://www.w3.org/TR/css-color-5/
+
+<!-- flesh this out more -->
+
+## Definitions
+
+### Color
+
+> Note that channel values are stored as specified, maintaining precision where
+> possible, even when the values are out-of-gamut for the given [color space][].
+
+A *color* is an object with several parts:
+
+* A string [*color space*](#color-space)
+
+* An ordered list of *channel* values as defined by that [color space][].
+
+* A numeric *alpha* value which can be safely clamped in the `0-1` or `0%-100%`
+  range. Values outside that range are allowed, but meaningless.
+
+* A boolean *is-legacy* to indicate a [legacy color][].
+
+[legacy color]: #legacy-color
+[color space]: #color-space
+
+### Legacy Color
+
+> Both Sass and CSS have similar legacy behavior that relies on all colors
+> being interchangeable as part of a shared `srgb` color space. While the new
+> color formats will opt users into new default behavior, some legacy color
+> formats behave differently for the sake of backwards-compatibility.
+
+Colors that are defined using the CSS color names, hex syntax, `rgb()`,
+`rgba()`, `hsl()`, `hsla()`, or `hwb()` -- along with colors that result from
+legacy interpolation -- are considered *legacy colors*. All legacy colors are
```

Comment:
I'm realizing now that "colors returned by color functions whose color inputs are all legacy" doesn't work, because you don't want `color.to-space()` to return a legacy color if you convert to a color space that otherwise never supports legacy colors.

This raises another interesting question—is the *only* behavioral distinction between the `rgb` and `srgb` spaces that `rgb` can be legacy while `srgb` never can? Should that mean that we automatically change any non-legacy `rgb` color into `srgb`? Or should we just say that only `srgb` is an actual named color space, and it just may or may not be legacy?